elements/pfe-textinput/src/pfe-textinput.js

```diff
@@ -0,0 +1,202 @@
+import PFElement from "../../pfelement/dist/pfelement.js";
+import "./polyfills--pfe-textinput.js";
+
+// @IE11
+// watching for addition and removal of nodes so
+// we can make sure we have the correct light DOM
+// and so we can set the _lightInput property
+const parentObserverConfig = {
+  childList: true
+};
+
+// watching for changes on the _lightInput so we can
+// update input attributes in our shadow DOM when the _lightInput
+// changes
+const lightInputObserverConfig = {
+  attributes: true
+};
+
+// list of attributes that we DO NOT want to pass from
+// the _lightInput to our shadow DOM input. For example,
+// the style attribute could ruin our encapsulated styles
+// in the shadow DOM
+const denylistAttributes = ["style", "tabindex"];
+
+class PfeTextinput extends PFElement {
+  static get tag() {
+    return "pfe-textinput";
+  }
+
+  get schemaUrl() {
+    return "pfe-textinput.json";
+  }
+
+  get templateUrl() {
+    return "pfe-textinput.html";
+  }
+
+  get styleUrl() {
+    return "pfe-textinput.scss";
+  }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Content;
+  }
+
+  constructor() {
+    super(PfeTextinput, { type: PfeTextinput.PfeType });
+
+    this._lightInput = null;
+    this._internalInput = null;
+    this._setFocus = this._setFocus.bind(this);
+    this._keyupListener = this._keyupListener.bind(this);
+
+    this._init = this._init.bind(this);
+
+    this._lightInputObserver = new MutationObserver(mutationsList => {
```

Comment:
@kylebuch8 What would you think of something like this going in the base class? cc @mwcz 

If not there, maybe it could be one of the JS mixins we were talking about?  This seems very reusable.

Reply:
I think it's a good idea. Just not anything I'd like to tackle in this PR.